core/src/main/java/org/apache/carbondata/core/keygenerator/directdictionary/timestamp/TimeStampDirectDictionaryGenerator.java

```diff
@@ -117,15 +117,24 @@ private TimeStampDirectDictionaryGenerator() {
    * @return dictionary value
    */
   @Override public int generateDirectSurrogateKey(String memberStr) {
-    SimpleDateFormat timeParser = new SimpleDateFormat(CarbonProperties.getInstance()
-        .getProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT,
-            CarbonCommonConstants.CARBON_TIMESTAMP_DEFAULT_FORMAT));
+    String timeString;
+    String formatString;
+    if (memberStr.contains(CarbonCommonConstants.COLON_SPC_CHARACTER)){
```

Comment:
It is not a good implement to append format to input data.
No need to combine data with data format to a new member string.